aptos-move/aptos-vm-types/src/environment.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_types::{
+    chain_id::ChainId,
+    on_chain_config::{
+        ConfigurationResource, Features, OnChainConfig, TimedFeatureOverride, TimedFeatures,
+        TimedFeaturesBuilder,
+    },
+    state_store::StateView,
+    vm::configs::{aptos_prod_vm_config, get_timed_feature_override},
+};
+use move_vm_runtime::config::VMConfig;
+use std::sync::Arc;
+
+/// A runtime environment which can be used for VM initialization and more.
+#[derive(Clone)]
+pub struct Environment {
+    pub chain_id: ChainId,
+
+    // Note: if features and timed features change, VM config needs re-initialization.
```

Comment:
is this maybe an argument to have accessor methods and hide changes with initialization APIs?